administrator/components/com_workflow/resources/scripts/app/WorkflowGraphApi.es6.js

```diff
@@ -0,0 +1,183 @@
+import Event from './Event.es6.js';
+
+class WorkflowGraphApi {
+  constructor() {
+    const options = Joomla.getOptions('com_workflow', {});
+    if (!options.apiBaseUrl) {
+      throw new TypeError('Workflow API baseUrl is not defined');
+    }
+    this.baseUrl = options.apiBaseUrl;
+
+    if (!options.extension) {
+      throw new TypeError('Workflow API extension is not defined');
+    }
+    this.extension = options.extension;
+    this.csrfToken = Joomla.getOptions('csrf.token');
+    if (!this.csrfToken) {
+      console.warn('CSRF token not found');
+    }
+  }
+
+  /**
+   * Generic request method with better error handling
+   */
+  async makeRequest(url, options = {}) {
+    const defaultOptions = {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Requested-With': 'XMLHttpRequest'
+      },
+    };
+
+    // add csrf for all request as data [token] = 1
+    if (this.csrfToken) {
+      if (!options.data) {
+        options.data = {};
+      }
+      options.data[this.csrfToken] = '1';
+    }
+
+    const config = { ...defaultOptions, ...options };
+
+    if (options.data instanceof FormData) {
+      delete config.headers['Content-Type'];
+    }
+    return new Promise((resolve, reject) => {
+      Joomla.request({
+        url: `${this.baseUrl}${url}&extension=${this.extension}`,
+        ...config,
+        onSuccess: (response) => {
+          resolve(response);
+        },
+        onError: (xhr) => {
+          let message = 'Network error';
+          try {
+            const errorData = JSON.parse(xhr.responseText);
+            message = errorData.message || message;
+          } catch (e) {
+            message = xhr.statusText || message;
+          }
+          reject(new Error(message));
+        }
+      });
+    });
+  }
+
+  async getWorkflow(id) {
```

Comment:
Could you work with some function comments to describe what the function does and what parameters are expected etc. (also for the other functions)